### PR TITLE
Chef Love

### PR DESCRIPTION
**Pull request by Kurfursten**
The chef has needed some love for a long time. Notably, the fact that all of his crates at cargo are better for bored cargo techs than the chef himself. This is the first of a few PRs from me aimed at reinvigorating the chef.
- Re-purposed the "food crate" into the "basic cooking crate", a good source of loads of flour. Now somewhat more competitive with just growing it yourself next door, since you can get Cargo to send a big load straight to your doorstep and focus on other things.
- Three new crates: Fruit Crate (contains a large, random assortment of fruits), Exotic Garnishes (six random condiments ranging from salt to honey), Trophy Meats (all sorts of fun meats for the rarer recipes that never get made) . The latter two require Kitchen access.
- New recipes for Malt Vinegar and Sprinkles. They use Universal Enzyme (mixed with ethanol and sugar respectively).
- Dropper refactor
- New premium item at the kitchen vendor: baster. It's a big dropper. 
  baster
- Droppers (and children) can now be used directly on the microwave to add reagents.
- Removed the non-exotic seed crate. It was redundant with the vendor recharge pack.
- Fixed a bug in which, although some hydroponics crates had access requirements, they did not actually use a crate which could use access requirements. (e.g.: bees)
- Removed hydroponics access from some harmless crates

### CODE REVIEW

**Comment by SkowronX** [comment]
>Trophy Meats

:+1::+1::+1:

**Comment by FudgePucker** [comment]
Do the mining mob meats contain anything special?

**Comment by SkowronX** [comment]
As far as I know, theyre just renamed base meat.

**Comment by Kurfursten** [comment]
The mining mob meats are just base meat, they're not used in any recipes. If someone adds recipes for them (please do), I'll add them to the trophy meats. That's mostly spriting, though.

**Comment by FudgePucker** [comment]
Maybe have basklisk meat have Cryoxadone and Cold Sauce. Goliath meat could have a cocktail of healing chems, such as DD tricord and others also a ton of nutrients. Hivelord meat should have little to no nutrients, but have Peridaxon because of its regenitive abilities.

A fun recipe might be miners delight, a slice of goliath meat, hivelord remains, a sheet of plasma, and a diamond ore would be the recipe.

**Comment by Kurfursten** [comment]
None of that is really in scope for this PR. That being said, I don't think turning every single mine enemy into a healthpack is wise.

**Comment by FudgePucker** [comment]
Have large amount of toxin damage from eating raw meat and a chance to get a random virus. (Why isn't this a thing?).
Goliaths are the hardest mobs outside of admin bus or away missions, the only other chemicals it can have in its meat that would explain its robustness is creatine, and I don't think we would want another source of that shit, basklisks I can understand if you would want to remove the cyroxedone, hivelord meat isn't that overpowered, hivelord remains can fix everything exept death. There should be a way to get that chem outside of xenobiology or xenoarchology, instead of a random chance to get it there would be a somewhat reliable source. It would be a useful barganing chip for bottles of hyperzine from the chemist.

**Comment by Kurfursten** [comment]
Miners have a truly crazy amount of bargaining tools already. They can run the Cargo console, they astronomical wealth, they can produce all the vendor coins they want, they have the tools and time to explore space for goodies like the pirate ship, they have a whole vendor of dedicated toys including jetpacks, jaunters, and guns and have great bargaining power over science and mechanics (who also have chem dispensers). If you can't leverage this to get hyperzine, you might need to work on your social skills.

I don't think mining creatures need to have chemicals innately present in them. I am interested in bringing more variety to the chef with new menu options. Of course, chemicals in finished meals (e.g.: tricordrazine in plump helmet pie) is always on the table.

**Comment by Kurfursten** [comment]
@clusterfack @N3X15  No conflicts here, time to merge?

I, for one, welcome our culinary future

File: code/game/machinery/kitchen/microwave.dm [hunk 1]
```diff
@@ -173,7 +173,8 @@
 				"<span class='notice'>You add \the [O] to \the [src].</span>")
 	else if(istype(O,/obj/item/weapon/reagent_containers/glass) || \
 	        istype(O,/obj/item/weapon/reagent_containers/food/drinks) || \
-	        istype(O,/obj/item/weapon/reagent_containers/food/condiment) \
+	        istype(O,/obj/item/weapon/reagent_containers/food/condiment) || \
```
Comment (PJB3005) [inline code/game/machinery/kitchen/microwave.dm]: Make this into an `is_type_in_list()`

File: code/modules/reagents/reagent_containers/dropper.dm [hunk 2]
```diff
@@ -19,110 +18,112 @@
...
-
-						user << "<span class='notice'>You transfer [trans] units of the solution.</span>"
-						if (src.reagents.total_volume<=0)
-							filled = 0
-							icon_state = "dropper[filled]"
-						return
+/obj/item/weapon/reagent_containers/dropper/update_icon()
+	icon_state = "dropper[(reagents.total_volume ? 1 : 0)]"
 
+/obj/item/weapon/reagent_containers/dropper/afterattack(obj/target, mob/user , flag)
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/dropper.dm]: Should be `.../afterattack(atom/target, mob/user, proximity_flag, click_parameters)`

File: code/modules/reagents/reagent_containers/dropper.dm [hunk 3]
```diff
@@ -19,110 +18,112 @@
...
+	if(!user.Adjacent(target))
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/dropper.dm]: Use `proximity_flag` instead.

File: code/modules/reagents/reagent_containers/dropper.dm [hunk 4]
```diff
@@ -19,110 +18,112 @@
...
+
+				var/obj/item/safe_thing = victim.get_body_part_coverage(EYES)
+
+				if(safe_thing)
+					if(!safe_thing.reagents)
+						safe_thing.create_reagents(100)
+					trans = src.reagents.trans_to(safe_thing, amount_per_transfer_from_this)
+
+					for(var/mob/O in viewers(world.view, user))
+						O.show_message(text("<span class='danger'>[] tries to squirt something into []'s eyes, but fails!</span>", user, target), 1)
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/dropper.dm]: `visible_message()`

File: code/modules/reagents/reagent_containers/dropper.dm [hunk 5]
```diff
@@ -19,110 +18,112 @@
...
+					spawn(5)
+						src.reagents.reaction(safe_thing, TOUCH)
+					user << "<span class='notice'>You transfer [trans] units of the solution.</span>"
+					update_icon()
+					return
+			for(var/mob/O in viewers(world.view, user))
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/dropper.dm]: `visible_message()`

File: code/modules/reagents/reagent_containers/dropper.dm [hunk 6]
```diff
@@ -19,110 +18,112 @@
...
-						if(reagents.has_reagent(bad_reagent))
-							badshit += reagents_to_log[bad_reagent]
-					if(badshit.len)
-						var/hl="<span class='danger'>([english_list(badshit)])</span>"
-						message_admins("[user.name] ([user.ckey]) added [trans]U to \a [target] with [src].[hl] (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)")
-						log_game("[user.name] ([user.ckey]) added [trans]U to \a [target] with [src].")
+		var/trans = 0
+
+		if(ismob(target))
+			if(istype(target , /mob/living/carbon/human))
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/dropper.dm]: `ishuman()`

File: code/modules/reagents/reagent_containers/food/condiment.dm [hunk 7]
```diff
@@ -267,3 +267,11 @@
 		..()
 		reagents.add_reagent("vinegar", 50)
 
+/obj/item/weapon/reagent_containers/food/condiment/exotic
+	name = "Exotic Bottle"
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/food/condiment.dm]: `\improper`

File: code/modules/reagents/reagent_containers/dropper.dm [hunk 8]
```diff
@@ -19,110 +18,112 @@
...
-			var/trans = target.reagents.trans_to(src, amount_per_transfer_from_this)
+		update_icon()
 
-			user << "<span class='notice'>You fill the dropper with [trans] units of the solution.</span>"
+	return
 
-			filled = 1
-			icon_state = "dropper[filled]"
+/obj/item/weapon/reagent_containers/dropper/baster
+	name = "Baster"
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/dropper.dm]: `\improper`

File: code/modules/reagents/reagent_containers/food/condiment.dm (hunk 7 above)
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/food/condiment.dm]: `\improper`

File: code/modules/reagents/reagent_containers/food/condiment.dm [hunk 9]
```diff
@@ -267,3 +267,11 @@
 		..()
 		reagents.add_reagent("vinegar", 50)
 
+/obj/item/weapon/reagent_containers/food/condiment/exotic
+	name = "Exotic Bottle"
+	desc = "If you can see this label, something is wrong."
+	//~9% chance of anything but special sauce, which is .09 chance
+	var/list/possible_condiments = list("enzyme"=10,"blackpepper"=10,"vinegar"=10,"sodiumchloride"=10,"cinnamon"=10,"chefspecial"=1,"frostoil"=10,"soysauce"=10,"capsaicin"=10,"honey"=10,"ketchup"=10,"coco"=10)
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/food/condiment.dm]: Make this a global list or something.

File: code/modules/reagents/reagent_containers/food/condiment.dm (hunk 7 above)
Comment (Kurfursten) [inline code/modules/reagents/reagent_containers/food/condiment.dm]: This (and the baster below) are actually better off just lower case entirely to follow naming convention

File: code/modules/reagents/reagent_containers/dropper.dm (hunk 3 above)
Comment (Kurfursten) [inline code/modules/reagents/reagent_containers/dropper.dm]: In this case just flag

File: code/modules/reagents/reagent_containers/food/condiment.dm (hunk 7 above)
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/food/condiment.dm]: Then make them lowercase?

File: code/modules/reagents/reagent_containers/dropper.dm (hunk 3 above)
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/dropper.dm]: See my above comment.

You should always keep the same argument names, no exceptions.

File: code/modules/reagents/reagent_containers/food/condiment.dm (hunk 7 above)
Comment (Kurfursten) [inline code/modules/reagents/reagent_containers/food/condiment.dm]: I wasn't arguing! You don't have to be defensive. I just don't have time to work on more code for a few more hours.

File: code/modules/reagents/reagent_containers/dropper.dm (hunk 4 above)
Comment (Kurfursten) [inline code/modules/reagents/reagent_containers/dropper.dm]: I swear I am about one more refactor PR from just going through the code to remove every instance of the garbage so I don't have to fix it over and over.

File: code/modules/reagents/reagent_containers/dropper.dm [hunk 10]
```diff
@@ -19,110 +18,112 @@
...
+			else
+				M.LAssailant = user
+
+		trans = src.reagents.trans_to(target, amount_per_transfer_from_this)
+		user << "<span class='notice'>You transfer [trans] units of the solution.</span>"
+		update_icon()
+
+		// /vg/: Logging transfers of bad things
+		if(isobj(target))
+			if(istype(reagents_to_log) && reagents_to_log.len && target.log_reagents)
```
Comment (Kurfursten) [inline code/modules/reagents/reagent_containers/dropper.dm:87]: How did this compile? log_reagents shouldn't be possible for an atom, it's a property of object

**Comment by Kurfursten** [comment]
@N3X15 Pls gib merge

**Comment by PJB3005** [comment]
Looks good.

**Pull request merged**